recipes/systemd_service.rb

```diff
@@ -15,20 +15,57 @@ class ::Chef::Recipe
   ['debian'] => ['debian', 'default', 'chef-client']
 )
 
+timer = node['chef_client']['systemd']['timer']
+
+exec_options = if timer
+                 '-c $CONFIG $OPTIONS'
+               else
+                 '-c $CONFIG -i $INTERVAL -s $SPLAY $OPTIONS'
+               end
+
 template '/etc/systemd/system/chef-client.service' do
   source 'systemd/chef-client.service.erb'
   mode '644'
-  variables(client_bin: client_bin, sysconfig_file: "/etc/#{conf_dir}/#{env_file}")
-  notifies :restart, 'service[chef-client]', :delayed
+  variables(
+    client_bin: client_bin,
+    sysconfig_file: "/etc/#{conf_dir}/#{env_file}",
+    type: (timer ? 'oneshot' : 'simple'),
+    exec_options: exec_options,
+    restart_mode: (timer ? nil : 'on-failure')
+  )
+  notifies :restart, 'service[chef-client]', :delayed unless node['chef_client']['systemd']['timer']
 end
 
 template "/etc/#{conf_dir}/#{env_file}" do
   source "#{dist_dir}/#{conf_dir}/chef-client.erb"
   mode '644'
-  notifies :restart, 'service[chef-client]', :delayed
+  notifies :restart, 'service[chef-client]', :delayed unless node['chef_client']['systemd']['timer']
 end
 
+service_actions = [:enable]
+service_actions << :start unless node['chef_client']['systemd']['timer']
+
 service 'chef-client' do
   supports status: true, restart: true
-  action [:enable, :start]
+  action service_actions
+end
+
+systemd_unit 'chef-client.timer' do
```

Comment:
good point, I'll fix that. In general it seems that most cookbooks are not "cleaning" resource when they are not necessary but in this case it would clearly make sense.